packages/rpc/src/rpc-default-config.ts

```diff
@@ -2,9 +2,9 @@ import type { createSolanaRpcApi } from '@solana/rpc-api';
 
 import { createSolanaJsonRpcIntegerOverflowError } from './rpc-integer-overflow-error';
 
-export const DEFAULT_RPC_CONFIG: Partial<Parameters<typeof createSolanaRpcApi>[0]> = {
+export const DEFAULT_RPC_CONFIG: Partial<NonNullable<Parameters<typeof createSolanaRpcApi>[0]>> = {
     defaultCommitment: 'confirmed',
     onIntegerOverflow(methodName, keyPath, value) {
         throw createSolanaJsonRpcIntegerOverflowError(methodName, keyPath, value);
     },
-};
+} as const;
```

Comment:
This prevents us from adding properties here that are _not_ part of the config type.